pkg/apply/desiredset_crud.go

```diff
@@ -51,9 +52,16 @@ func (o *desiredSet) get(nsed bool, namespace, name string, client dynamic.Names
 	return client.Get(o.ctx, name, v1.GetOptions{})
 }
 
-func (o *desiredSet) delete(nsed bool, namespace, name string, client dynamic.NamespaceableResourceInterface, force bool) error {
-	if o.noDelete && !force {
-		return nil
+func (o *desiredSet) delete(nsed bool, namespace, name string, client dynamic.NamespaceableResourceInterface, force bool, gvk schema.GroupVersionKind) error {
+	if !force {
+		if o.noDelete {
+			return nil
+		}
+		for key := range o.noDeleteGVK {
```

Comment:
You can just do 
if _, ok := o.noDeleteGVK[key]; ok {
  return nil
}

Reply:
_facepalm_ thank you. That's what I get for quick refactoring.